### PR TITLE
Fixes #18721: Allow defining port in policy_server.dat

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/18721

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

File: techniques/system/common/1.0/update.st [hunk 1]
```diff
@@ -78,6 +86,9 @@ bundle common server_info
 
   classes:
       "policy_server_file_exists" expression => fileexists("${policy_server_file}");
+      "policy_server_has_port"    expression => strcmp(length("policy_server_temp3"), "2");
+      "pass3" expression => "pass2";
+      "pass2" expression => "any";
```
Comment (ncharles) [inline techniques/system/common/1.0/update.st:91]: these classes become global: you must not put pass2 in a bundle common

**Comment by peckpeck** [comment]
PR updated with a new commit

File: techniques/system/common/1.0/update.st [hunk 2]
```diff
@@ -63,9 +63,17 @@ bundle common server_info
...
+      "policy_server_temp"   string => readfile("${policy_server_file}", 2048);
+      # trim space in policy_server.dat
+      "policy_server_temp2"  string => regex_replace("${policy_server_temp}", " ", "", "g");
+      # separate host and port
+      "policy_server_temp3"  slist  => string_split("${policy_server_temp2}", ":", "2");
+      "policy_server"        string => nth("policy_server_temp3", "0");
+    # pass3 is needed otherwise system_common.community_port is evaluated during pass2 where policy_server_has_port is always false
+    server_info_pass3.policy_server_has_port::
+      "policy_server_port"   string => nth("policy_server_temp3", "1");
+    server_info_pass3.!policy_server_has_port::
```
Comment (ncharles) [inline techniques/system/common/1.0/update.st:75]: as common.st is imported before update.st, it eagerly resolves variables in common.st
i'd rather have this logic in the same file as common.st (before bundle system_common) to have something consistent with failsafe.cf

Comment (peckpeck) [reply techniques/system/common/1.0/update.st:75]: there is no system_common in failsafe.cf

File: techniques/system/common/1.0/common.st [hunk 3]
```diff
@@ -3,7 +3,8 @@ bundle common system_common {
   vars:
     "directiveId"           string => "&TRACKINGKEY&";
     "sharedfiles_folder"    string => "&SHARED_FILES_FOLDER&";
-    "community_port"        string => "&COMMUNITYPORT&";
+    "cf_serverd_port"       string => "&COMMUNITYPORT&";
```
Comment (Fdall) [inline techniques/system/common/1.0/common.st:6]: &COMMUNITYPORT& is also used in cf-serverd.st shouldn't it be changed there as well?

Comment (peckpeck) [reply techniques/system/common/1.0/common.st:6]: this is just a name change, &COMMUNITYPORT& already contains the proper port

**Review by Fdall — APPROVED** [review]

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**